lib/buffer.js

```diff
@@ -266,7 +266,9 @@ Buffer.alloc = function alloc(size, fill, encoding) {
     // be interpreted as a start offset.
     if (typeof encoding !== 'string')
       encoding = undefined;
-    return createUnsafeBuffer(size).fill(fill, encoding);
+    const ret = createUnsafeBuffer(size);
+    if (fill_(ret, fill, encoding))
```

Comment:
nit: `if (fill_(ret, fill, encoding) > 0) {` ? To avoid an unnecessary ToBoolean.

Reply:
@targos yup, will fix while landing